File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        
        for i, val in enumerate(nums):
            comp = target - val
            if comp in seen:
                return [i, seen[comp]]
            seen[val] = i
        
        
            

File: 0001-two-sum/0001-two-sum.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;
        
        // O(n)
        for(int i = 0; i < nums.size(); i++) {
            int comp = target - nums[i];
            if (hashMap.contains(nums[i])) {
                int res = hashMap[nums[i]];
                return vector<int>{i, res};
            }
            hashMap[comp] = i;
        }
        
        return vector<int>{};
        
    }
};